pkg/controllers/vdb/restart_reconciler.go

```diff
@@ -336,12 +326,22 @@ func (r *RestartReconciler) restartPods(ctx context.Context, pods []*PodFact) (c
 
 // removePodsWithClusterUpState will see if the pods in the down list are
 // down according to the cluster state. This will return a new pod list with the
-// pods that aren't considered down removed.
+// pods that aren't considered down removed. It exists early if there is no
+// cluster quorum
 func (r *RestartReconciler) removePodsWithClusterUpState(ctx context.Context, pods []*PodFact) ([]*PodFact, ctrl.Result, error) {
 	clusterState, res, err := r.fetchClusterNodeStatus(ctx, pods)
 	if verrors.IsReconcileAborted(res, err) {
 		return nil, res, err
 	}
+	// If cluster does not have quorum, we requeue to
+	// restart the entire cluster
+	if !r.doesDBHaveQuorum(clusterState) {
```

Comment:
We are kind of abusing this function. For example, this new check isn't to remove pods with cluster up state. I suggest we do a bit of a refactor. Maybe pass in the clusterState into this function and move the fetchClusterNodeStatus to a calling function.